### ISSUE TITLE
[BP-640] v1.30.0 - Cookie must be secure in order to use the SameSite None directive.

### ISSUE THREAD

**Issue by j-stam** [buckaroo-it/Magento2#64]
Using: Magento 2.4.1 and module version v1.30.0

I encountered the following error: `Cookie must be secure in order to use the SameSite None directive.` with the FixSession.php plugin where the SameSite cooke is changed to None.

The SameSite parameter is changed before my cookie is set to be secure, which triggers this issue.

This patch fixes the issue.

```
diff --git a/Model/Plugin/FixSession.php b/Model/Plugin/FixSession.php
index 0b3e9d7..76decd8 100644
--- a/Model/Plugin/FixSession.php
+++ b/Model/Plugin/FixSession.php
@@ -53,8 +53,8 @@ class FixSession
         if ($metadata && method_exists($metadata, 'getSameSite') && ($name == 'PHPSESSID')) {
             //$this->logger->addDebug(__METHOD__ . '|1|' . var_export([$name, $value, $metadata->getSameSite()], true));
             if ($metadata->getSameSite() != 'None') {
-                $metadata->setSameSite('None');
                 $metadata->setSecure(true);
+                $metadata->setSameSite('None');
             }
         }
         return [$name, $value, $metadata];
```

**Comment by Buckaroo-Rens** [comment]
Hello @j-stam, a simmular issue should have been solved in 1.30 but we will research your request again. Please take a close look at: https://github.com/buckaroo-it/Magento2/commit/ce7a0c85d1899c6e44af371e4ef8aad329cb3f30

**Comment by j-stam** [comment]
@Buckaroo-Rens 

The fix made in ce7a0c8 isn't working for me because the cookie that is being modified isn't secure, changing the order of setting secure and samesite (as seen in the patch i provided) fixes this issue because secure is forced before samesite is changed.

**Comment by Buckaroo-Rens** [comment]
Issue is fixed and released in v1.32
